cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -339,10 +337,19 @@ _CCCL_DEVICE void memcpy_async_aligned(void* dst, const void* src, unsigned int
   for (unsigned int offset = threadIdx.x * ldgsts_size_and_align; offset < bytes_to_copy;
        offset += BlockThreads * ldgsts_size_and_align)
   {
-    __pipeline_memcpy_async(
-      static_cast<char*>(dst) + offset, static_cast<const char*>(src) + offset, ldgsts_size_and_align);
+    asm volatile(
```

Comment:
This looks correct, I see that the pointers are marked as `__restrict__`

Should we also add `_CCCL_RESTRICT` to the function signatures?